src/moderation/moderation.service.ts

```diff
@@ -27,4 +27,56 @@ export class ModerationService {
                     !blockedAccountIds.has(post.repostedBy.id)),
         );
     }
+
+    async filterUsersForPost(
+        userIds: number[],
+        post: Post,
+        repostedBy?: number,
+    ) {
+        // Map user ids to their corresponding account ids
+        const userAccountMap = new Map<number, number>();
+
+        for (const row of await this.db('users')
+            .whereIn('id', userIds)
+            .select('id', 'account_id')) {
+            userAccountMap.set(row.id, row.account_id);
+        }
+
+        // If the post has been reposted by an account, check if the reposter
+        // has blocked the author, and if so, filter out everybody but the
+        // reposter
+        if (repostedBy) {
+            const authorHasBlockedReposter = await this.db('blocks')
+                .where('blocker_id', post.author.id)
+                .andWhere('blocked_id', repostedBy)
+                .first();
+
+            if (authorHasBlockedReposter) {
+                return [userAccountMap.get(repostedBy)];
```

Comment:
I think this can return `[undefined]` - an explicit return type in the function sig would have caught this!

Reply:
Nice catch! Updated in https://github.com/TryGhost/ActivityPub/pull/572/commits/ef06d1aaac15e7fd9631478e7e96b3874073c6f5